airbyte-integrations/connectors/source-microsoft-onedrive/integration_tests/abnormal_state.json

```diff
@@ -6,10 +6,13 @@
         "name": "test_csv"
       },
       "stream_state": {
-        "_ab_source_file_last_modified": "2023-12-23T06:49:25.000000Z_Test_folder_2/TestFileOneDrive.csv",
         "history": {
-          "Test_folder_2/TestFileOneDrive.csv": "2023-12-23T06:49:25.000000Z"
-        }
+          "https://airbyte179.sharepoint.com/Shared%20Documents/Test_folder/TestFile.csv": "2023-11-17T13:52:35.000000Z",
```

Comment:
why file name changed to full url?

Reply:
Should it be `Shared%20Documents/Test_folder/TestFile.csv` to be consistent in the keys name

Reply:
@lazebnyi `https://airbyte179.sharepoint.com/Shared%20Documents/` [is the `web_url` parameter](https://github.com/airbytehq/airbyte/pull/35849/files#diff-d9c193522aeda1720ba05cd5d2a8c480b641a137bae83d5029747393adb5da88R222) of the shared item. It cannot be parsed as different drive items may have files with the same paths, so the whole URL is the only choice.

Reply:
Alright, I understand. In that case, we'll need to update the implementation for user space files and also return their full paths in the state.

Reply:
Do you mean the URL of the file? But the current path is defined by the drive name and folder path from the config.